classes/blocks/class-media.php

```diff
@@ -23,41 +23,22 @@ class Media extends Base_Block {
 	 */
 	const BLOCK_NAME = 'media_video';
 
-	/**
-	 * Register shortcake shortcode.
-	 *
-	 * @param array  $attributes Shortcode attributes.
-	 * @param string $content   Content.
-	 *
-	 * @return mixed
-	 */
-	public function add_block_shortcode( $attributes, $content ) {
-
-		$attributes = shortcode_atts(
-			[
-				'video_title'      => '',
-				'description'      => '',
-				'youtube_id'       => '',
-				'video_poster_img' => '',
-			],
-			$attributes,
-			'shortcake_media_video'
-		);
-
-		return $this->render( $attributes );
-	}
-
 	/**
 	 * Media constructor.
 	 */
 	public function __construct() {
-		add_shortcode( 'shortcake_media_video', [ $this, 'add_block_shortcode' ] );
-
 		register_block_type(
 			'planet4-blocks/media-video',
 			[
 				'editor_script'   => 'planet4-blocks',
-				'render_callback' => [ $this, 'render' ],
+				'render_callback' => static function ( $attributes, $content ) {
+					if ( '' === trim( $content ) ) {
+						$json = wp_json_encode( [ 'attributes' => $attributes ] );
+						return '<div data-render="planet4-blocks/media-video" data-attributes="' . htmlspecialchars( $json ) . '"></div>';
+					} else {
```

Comment:
No need for the else here

Reply:
A bit more straightforward to invert the condition here, as less will be nested.
```
if ( '' !== trim($content)) {
    return $content;
}

$json = wp_json_encode( [ 'attributes' => $attributes ] );

return '<div data-render="planet4-blocks/media-video" data-attributes="' . htmlspecialchars( $json ) . '"></div>';
```